extensions/objects/templates/math-editor.component.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for math editor.
+ */
+
+// Every editor directive should implement an alwaysEditable option. There
+// may be additional customization options for the editor that should be passed
+// in via initArgs.
+
+angular.module('oppia').component('mathEditor', {
+  bindings: {
+    value: '='
+  },
+  template: require('./math-editor.component.html'),
+  controller: [function() {
+    const ctrl = this;
+
+    ctrl.assignIdToGuppyDiv = function() {
+      // Dynamically assigns a unique id to the guppy-div
+      var divId = 'guppy_' + Math.floor(Math.random() * 100000000);
+      $('.guppy-div').attr('id', divId);
+      return divId;
+    };
+
+    ctrl.createGuppyInstance = function(divId: string) {
+      return new Guppy(divId, {});
+    };
+
+    ctrl.$onInit = function() {
+      ctrl.alwaysEditable = true;
+      var divId = ctrl.assignIdToGuppyDiv();
+      var guppyInstance = ctrl.createGuppyInstance(divId);
+      guppyInstance.event('change', (e) => {
```

Comment:
Should we add the keydown event listener here instead of inside the script tag? (not sure whether that will work!)
Asking such that we have less code there inside the script tag and we can use $document)

Reply:
Valid point -- but the guppy.html file is common for all uses of guppy in the codebase. Currently, there is the MathExpressionInput interaction which uses guppy, also, soon, I plan on adding the new interactions that will also use guppy. So, the eventListener would have to be added to all of them separately. Adding it here instead, makes it global and easier to track.

Reply:
I'm a little confused. What is this conversation actually referring to and what is the resolution? I don't see any "keydown" in this PR.

Reply:
This is obsolete now. Before, I'd added an event listener that would hide the osk if the escape key is pressed but since we're not gonna use the osk on desktop anyways, this conversation is irrelevant.